File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        // two pointer approach 
        int i = 0;
        int j = height.size()-1;
        int ans = 0;
        while(i < j){
            
            int area = (min(height[i], height[j])) * (j - i);

            if(height[i] < height[j]){
                i++;
            }else {
                j--;
            }

            ans = max(ans, area);

        }
        return ans;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int i =0;
        int j = height.length -1;

        int maxArea = 0;

        while(i < j){

            
            int currArea = Math.min(height[i], height[j]) * (j-i);
            maxArea = Math.max(currArea, maxArea);

            
            if(height[i] > height[j]){
                j--;
            }else{
                i++;
            }
            
        }

        return maxArea;
        
    }
}